src/Utility/Hash.php

```diff
@@ -649,7 +643,7 @@ public static function check(array $data, string $path): bool
      * @return array Filtered array
      * @link https://book.cakephp.org/4/en/core-libraries/hash.html#Cake\Utility\Hash::filter
      */
-    public static function filter(array $data, $callback = ['self', '_filter']): array
+    public static function filter(array $data, callable|array $callback = ['self', '_filter']): array
```

Comment:
Does the `$callback` arg need to have `|array` because PHP can't always figure out if the argument is a valid callable?

Reply:
The problem is the default. The default is just an array that gets re-interpreted at runtime when used in the context of the function. Seems invalid to me.

Reply:
I changed this to `?callable` and created the default inside the method.